jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/OpenShellInContainerDialog.kt

```diff
@@ -78,7 +83,17 @@ class OpenShellInContainerDialog(
 
     override fun doOKAction() {
         super.doOKAction()
-        runExecCommand()
+        val task = tasks.selected() ?: throw IllegalStateException("Task not Selected")
+        launch {
+            val taskRoleFound: Boolean = EcsExecUtils.checkRequiredPermissions(project, container.service.clusterArn(), task)
```

Comment:
this kicks off a lot of AWS API calls. how do we provide a better experience here in the worst-case? what about caching?

Reply:
Most of the API calls are made in the scenario where task role is not found in the task definition and if the launch type is EC2, honestly will probably need to understand from the exec team the probability of such a scenario occurring, cause if its not too much im not sure if we should cache it?